plugins/postcss-cascade-layers/src/index.ts

```diff
@@ -93,10 +70,11 @@ const creator: PluginCreator<undefined> = () => {
 				if (hasLayerAtRuleAncestor(rule)) {
 					return;
 				}
+
 				rule.selectors = rule.selectors.map((selector) => {
 					// Needs `postcss-selector-parser` to insert `:not()` before any pseudo elements like `::after`
 					// This is a side track and can be fixed later.
-					return `${generateNot(layerCount)} ${selector}`;
+					return `${selector}${generateNot(layerCount)}`;
```

Comment:
- `:not(*) foo` assumes at least two levels in html.
- `foo:not(*)` does not.

Without a space it should work better.